File: JAVA/src/clases/Caballo.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package clases;

import clases_abstractas.Mamiferos;
import interfaces.IHabitat;
import interfaces.IReproducirse;

/**
 *
 * @author CHALA RAMIREZ
 */
public class Caballo extends Mamiferos implements IHabitat, IReproducirse {

    // Constructor con parametros
    public Caballo(String nombre, byte edad) {
        super(nombre, edad);
    }
    
    // Constructor sin parametros
    public Caballo() {
        super("", (byte) 12);
        this.nombre = "Pegaso";
        this.edad = 12;
    }
    
    // Constructor copia
    public Caballo(Caballo c) {
        super("", (byte) 12);
        this.nombre = c.nombre;
        this.edad = c.edad;
    }
    
    // Destructor
    public void close() {
        System.out.println("Destructor...");
    }

    // Metodo sobreescrito
    @Override
    public void amamantar() {
        System.out.println("El caballo amamanta a sus crias.");
    }

    // Metodos implementados de las interfaces
    @Override
    public void habitat() {
        System.out.println("De acuerdo a sus origenes, como tambien de manera global.");
    }

    @Override
    public void reproducirse() {
        System.out.println("Se reproducen sexualmente.");
    }
}

File: C#/C#/clases/Caballo.cs
using C_.clases_abstractas;
using C_.interfaces;
using System;

namespace C_.clases
{
    internal class Caballo : Mamiferos, IHabitat, IReproducirse
    {
        // Constructor con parametros
        public Caballo(String nombre, byte edad) : base(nombre, edad)
        {

        }

        // Constructor sin parametros
        public Caballo() : base("", (byte)12)
        {
            this.nombre = "Pegaso";
            this.edad = 12;
        }

        // Constructor copia
        public Caballo(Caballo c) : base("", (byte)12)
        {
            this.nombre = c.nombre;
            this.edad = c.edad;
        }

        // Destructor
        ~Caballo()
        {
            Console.WriteLine("Destructor...");
        }

        // Metodo sobreescrito
        public override void amamantar()
        {
            Console.WriteLine("El caballo amamanta a sus crias.");
        }

        // Metodos implementados de las interfaces
        public void habitat()
        {
            Console.WriteLine("De acuerdo a sus origenes, como tambien de manera global.");
        }

        public void reproducirse()
        {
            Console.WriteLine("Se reproducen sexualmente.");
        }
    }
}
